### PR TITLE
API 9766 Transformer Update

### PR DESCRIPTION
**Pull request by that-guy-is-it**

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ Immunization toFhir() {
         The primary source for an administered Immunization is true, as the provider of the
         record is the performer indicated below. See Data-Sources team for more info.
         */
-        .primarySource(Boolean.TRUE)
+        .primarySource(datamart.primarySource())
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:106]: The above comment is now obsolete

**Comment by jblefkove-va** [comment]
Now that `primarySource` is populated in all the payloads, you can make it a primitive boolean in `DatamartImmunization.java`

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/ImmunizationSamples.java [hunk 2]
```diff
@@ -34,6 +34,7 @@ public static Registration registration(String cdwId, String publicId) {
   public static class Datamart {
     @SneakyThrows
     public ImmunizationEntity entity(DatamartImmunization dm) {
+      dm.primarySource(true);
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/ImmunizationSamples.java:37]: This seems like a hack. Can we remove this and populate `primarySource` directly on the source data?

Comment (that-guy-is-it) [reply data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/ImmunizationSamples.java:37]: Ah Jesus, I am sorry about that one, I was testing a separate change and forgotten to properly fix this later. On it!

File: data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java [hunk 3]
```diff
@@ -17,10 +17,7 @@ void empty() {
                 .build()
                 .toFhir())
         .isEqualTo(
-            Immunization.builder()
-                .resourceType("Immunization")
-                .primarySource(Boolean.TRUE)
-                .build());
+            Immunization.builder().resourceType("Immunization").primarySource(false).build());
```
Comment (jblefkove-va) [inline data-query/src/test/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformerTest.java:20]: I think specifying the `resourceType` here is also redundant now

**Review by jblefkove-va — APPROVED** [review]
:shipit:

**Pull request merged**